test/graphql-api/utils/executeOperation.ts

```diff
@@ -51,21 +53,54 @@ export const executeOperation = async (
 
   const urlString = url.toString()
 
+  // Cannot pass files because of Puppeteer's limitation.
   const responsePromise = page.evaluate(
-    (url, method, query, variables) => {
+    (
+      url,
+      method,
+      query,
+      variables,
+      map,
+      fileContents: string[] | undefined,
+    ) => {
+      const getMultipartGraphQLBody = (
+        operations: string,
+        map: Record<string, string[]>,
+        fileContents: string[],
+      ): FormData => {
+        const body = new FormData()
+        body.append('operations', operations)
+        body.append('map', JSON.stringify(map))
+        const files = fileContents.map(
+          (f: string, i: number) => new File([f], `file${i}.txt`),
```

Comment:
No need to explicitly annotate `f` and `i`. `fileContents` is a type-safe array of strings (`string[]`). Those argument types will be inferred.